### PR TITLE
Add API for server to tell client to show message above hotbar

### PR DESCRIPTION
**Pull request by miozune**

### CODE REVIEW

**Comment by boubou19** [comment]
what's the difference with #7 ?

**Comment by miozune** [comment]
`printMessageAboveHotbar` works only with client. If server wants client to show message (e.g. block activation), server needs to send packet to tell client to do that. This PR contains the latter.

File: src/main/java/com/gtnewhorizon/gtnhlib/ClientProxy.java [hunk 1]
```diff
@@ -112,4 +115,17 @@ public void addMessageToChat(IChatComponent componentText) {
...
     }
+
+    @Override
+    public void sendMessageAboveHotbar(
+            EntityPlayerMP player,
+            IChatComponent chatComponent,
+            int displayDuration,
+            boolean drawShadow,
+            boolean shouldFade) {
+        if (FMLCommonHandler.instance().getEffectiveSide() == Side.CLIENT) {
```
Comment (Glease) [inline src/main/java/com/gtnewhorizon/gtnhlib/ClientProxy.java:126]: I can't think of a situation where you have a EntityPlayerMP but is on logical client.

File: src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java [hunk 2]
```diff
@@ -49,4 +54,18 @@ public void addErrorToChat(IChatComponent componentText) {}
...
+     * Sends packet from server to client that will display message above hotbar.
+     * @see ClientProxy#printMessageAboveHotbar
+     */
+    public void sendMessageAboveHotbar(
+            EntityPlayerMP player,
+            IChatComponent chatComponent,
+            int displayDuration,
+            boolean drawShadow,
+            boolean shouldFade) {
+        NetworkHandler.instance.sendTo(
```
Comment (Glease) [inline src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java:68]: As a precaution, please rule out fake player here.

Comment (draknyte1) [reply src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java:68]: Also remember TC doesn’t extend FakePlayer.

File: src/main/java/com/gtnewhorizon/gtnhlib/ClientProxy.java (hunk 1 above)
Comment (miozune) [reply src/main/java/com/gtnewhorizon/gtnhlib/ClientProxy.java:126]: Indeed, `EntityPlayerMP` requires `MinecraftServer` and `WorldServer` for instantiation

File: src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java (hunk 2 above)
Comment (miozune) [reply src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java:68]: sure

Comment (draknyte1) [reply src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java:70]: @miozune still doesn’t cover TC fake players.

Comment (miozune) [reply src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java:70]: Now I think, is this check really needed? For example GG does send message without FakePlayer check:
https://github.com/GTNewHorizons/GoodGenerator/blob/bd60bd3e96368d1fe726cf501ce1166b397d896a/src/main/java/goodgenerator/blocks/tileEntity/GTMetaTileEntity/NeutronSensor.java#L102

**Review by eigenraven — APPROVED** [review]
The check is useful for defensive coding to prevent crashes by api misuse, but not strictly necessary. We should just fix TC via mixins to make its fake player derive the FakePlayer class, not add workarounds in other mods

**Pull request merged**